pkg/client/client.go

```diff
@@ -182,16 +236,49 @@ func (c *KpmClient) ResolveDepsIntoMap(kclPkg *pkg.KclPkg) (map[string]string, e
 	}
 	var pkgMap map[string]string = make(map[string]string)
 	for _, d := range depMetadatas.Deps {
-		pkgMap[d.GetAliasName()] = d.GetLocalFullPath(kclPkg.HomePath)
+		pkgMap[d.GetAliasName()] = d.GetLocalFullPath()
 	}
 
 	return pkgMap, nil
 }
 
+// Get the search for the dependency.
+// 1. in the KCL_PKG_PATH: default is $HOME/.kcl/kpm
+// 2. in the vendor subdirectory of the current package.
+// 3. the dependency is from the local path.
+func (c *KpmClient) getDepStorePath(dep pkg.Dependency, isVendor bool) string {
+	if dep.IsFromLocal() {
+		return dep.GetLocalFullPath()
+	} else {
+		if isVendor {
+			return filepath.Join(dep.HomePath, "vendor", dep.GenDepFullName())
```

Comment:
The HomePath of dep is strange. The HomePath of kcl.Pkg is okay.

Reply:
fixed. The HomePath of dep has been removed.